### PR TITLE
BankingStage Refactor: Separate Consumer Module

### PR DESCRIPTION
**Pull request by apfitzge**
#### Problem
Splitting last major BankingStage component out of the top-level file.

`banking_stage.rs` is now primarly setup, control flow, and higher-level tests.

**STRONGLY RECCOMEND NOT USING GITHUB TO VIEW DIFF**
Try using this alias (or command):
```bash
alias git-move-diff='git rev-parse --abbrev-ref HEAD | git show --color-moved-ws=ignore-all-space -w --patch-with-stat  --color-moved'
```
(thanks @ryoqun)

#### Summary of Changes
Move `PreBalanceInfo` into `committer.rs`
Move `consume_buffered_packets` to new module: `consumer.rs`

Fixes #

### CODE REVIEW

File: core/src/banking_stage/consumer.rs [hunk 1]
```diff
@@ -0,0 +1,2074 @@
+use {
+    super::{
+        committer::{CommitTransactionDetails, Committer},
+        BankingStageStats,
```
Comment (apfitzge) [inline core/src/banking_stage/consumer.rs:4]: a remaining thing that *may* be nice to pull out of the top-level file...if for nothing else, to remove this circular dependencies.

**Comment by ryoqun** [comment]
> ```shell
> ```shell
> alias git-move-diff='git rev-parse --abbrev-ref HEAD | git show --color-moved-ws=ignore-all-space -w --patch-with-stat  --color-moved'
> ```
> (thanks @ryoqun)

well, there's no need for feeding `git show`'s stdin. it's not reading and is defaulting to `HEAD` because you gave no `<object>` on its args. sorry, i have to correct you to publicly show my decades' true love devoted to `git`. ;)

File: core/src/banking_stage/committer.rs [hunk 2]
```diff
@@ -1,5 +1,4 @@
 use {
-    super::PreBalanceInfo,
```
Comment (ryoqun) [inline core/src/banking_stage/committer.rs:2]: :heart:

**Comment by ryoqun** [comment]
Seems mindful selection/ordering was done for tests along with the move. nice.


<pre>$ git show --color-moved-ws=ignore-all-space -w --patch-with-stat  --color-moved | grep -E &apos;^[-].* fn |^.*[.]rs&apos;
<font color="#EF2929"><b> core/benches/banking_stage.rs</b></font>       |    6 +- 
<font color="#EF2929"><b> core/src/banking_stage.rs</b></font>           | 2073 +---------------------------------
<font color="#EF2929"><b> core/src/banking_stage/committer.rs</b></font> |   16 +-
<font color="#EF2929"><b> core/src/banking_stage/consumer.rs</b></font>  | 2074 +++++++++++++++++++++++++++++++++++
<font color="#EF2929"><b>diff --git a/core/benches/banking_stage.rs b/core/benches/banking_stage.rs</b></font>
<font color="#EF2929"><b>--- a/core/benches/banking_stage.rs</b></font>
<font color="#EF2929"><b>+++ b/core/benches/banking_stage.rs</b></font>
<font color="#EF2929"><b>diff --git a/core/src/banking_stage.rs b/core/src/banking_stage.rs</b></font>
<font color="#EF2929"><b>--- a/core/src/banking_stage.rs</b></font>
<font color="#EF2929"><b>+++ b/core/src/banking_stage.rs</b></font>
<font color="#EF2929"><b>-    fn </b></font>do_process_packets(
<font color="#EF2929"><b>-    pub fn </b></font>consume_buffered_packets(
<font color="#EF2929"><b>-    fn </b></font>execute_and_commit_transactions_locked(
<font color="#EF2929"><b>-    pub fn </b></font>process_and_record_transactions(
<font color="#EF2929"><b>-    fn </b></font>accumulate_execute_units_and_time(execute_timings: &amp;ExecuteTimings) -&gt; (u64, u64) {
<font color="#EF2929"><b>-    fn </b></font>process_transactions(
<font color="#EF2929"><b>-    fn </b></font>prepare_filter_for_pending_transactions(
<font color="#EF2929"><b>-    fn </b></font>filter_valid_transaction_indexes(valid_txs: &amp;[TransactionCheckResult]) -&gt; Vec&lt;usize&gt; {
<font color="#EF2929"><b>-    fn </b></font>filter_pending_packets_from_pending_txs(
<font color="#EF2929"><b>-    fn </b></font>process_packets_transactions(
<font color="#EF2929"><b>-    fn </b></font>sanitize_transactions(txs: Vec&lt;Transaction&gt;) -&gt; Vec&lt;SanitizedTransaction&gt; {
<font color="#EF2929"><b>-    fn </b></font>test_bank_prepare_filter_for_pending_transaction() {
<font color="#EF2929"><b>-    fn </b></font>test_bank_filter_valid_transaction_indexes() {
<font color="#EF2929"><b>-    fn </b></font>test_bank_process_and_record_transactions() {
<font color="#EF2929"><b>-    fn </b></font>test_bank_process_and_record_transactions_all_unexecuted() {
<font color="#EF2929"><b>-    fn </b></font>test_bank_process_and_record_transactions_cost_tracker() {
<font color="#EF2929"><b>-    fn </b></font>simulate_poh(
<font color="#EF2929"><b>-    fn </b></font>test_bank_process_and_record_transactions_account_in_use() {
<font color="#EF2929"><b>-    fn </b></font>test_process_transactions_returns_unprocessed_txs() {
<font color="#EF2929"><b>-    fn </b></font>execute_transactions_with_dummy_poh_service(
<font color="#EF2929"><b>-    fn </b></font>test_process_transactions_instruction_error() {
<font color="#EF2929"><b>-    fn </b></font>test_process_transactions_account_in_use() {
<font color="#EF2929"><b>-    fn </b></font>test_write_persist_transaction_status() {
<font color="#EF2929"><b>-    fn </b></font>generate_new_address_lookup_table(
<font color="#EF2929"><b>-    fn </b></font>store_address_lookup_table(
<font color="#EF2929"><b>-    fn </b></font>test_write_persist_loaded_addresses() {
<font color="#EF2929"><b>-    fn </b></font>setup_conflicting_transactions(
<font color="#EF2929"><b>-    fn </b></font>test_consume_buffered_packets() {
<font color="#EF2929"><b>-    fn </b></font>test_consume_buffered_packets_sanitization_error() {
<font color="#EF2929"><b>-    fn </b></font>test_consume_buffered_packets_interrupted() {
<font color="#EF2929"><b>-    fn </b></font>test_accumulate_execute_units_and_time() {
<font color="#EF2929"><b>diff --git a/core/src/banking_stage/committer.rs b/core/src/banking_stage/committer.rs</b></font>
<font color="#EF2929"><b>--- a/core/src/banking_stage/committer.rs</b></font>
<font color="#EF2929"><b>+++ b/core/src/banking_stage/committer.rs</b></font>
<font color="#EF2929"><b>diff --git a/core/src/banking_stage/consumer.rs b/core/src/banking_stage/consumer.rs</b></font>
<font color="#EF2929"><b>+++ b/core/src/banking_stage/consumer.rs</b></font>
</pre>



<pre><font color="#729FCF"></font>$ git show --color-moved-ws=ignore-all-space -w --patch-with-stat  --color-moved | grep -E &apos;^[+].* fn |^.*[.]rs&apos; 
<font color="#EF2929"><b> core/benches/banking_stage.rs</b></font>       |    6 +-
<font color="#EF2929"><b> core/src/banking_stage.rs</b></font>           | 2073 +---------------------------------
<font color="#EF2929"><b> core/src/banking_stage/committer.rs</b></font> |   16 +-
<font color="#EF2929"><b> core/src/banking_stage/consumer.rs</b></font>  | 2074 +++++++++++++++++++++++++++++++++++
<font color="#EF2929"><b>diff --git a/core/benches/banking_stage.rs b/core/benches/banking_stage.rs</b></font>
<font color="#EF2929"><b>--- a/core/benches/banking_stage.rs</b></font>
<font color="#EF2929"><b>+++ b/core/benches/banking_stage.rs</b></font>
<font color="#EF2929"><b>diff --git a/core/src/banking_stage.rs b/core/src/banking_stage.rs</b></font>
<font color="#EF2929"><b>--- a/core/src/banking_stage.rs</b></font>
<font color="#EF2929"><b>+++ b/core/src/banking_stage.rs</b></font>
<font color="#EF2929"><b>+    pub(crate) fn </b></font>simulate_poh(
<font color="#EF2929"><b>diff --git a/core/src/banking_stage/committer.rs b/core/src/banking_stage/committer.rs</b></font>
<font color="#EF2929"><b>--- a/core/src/banking_stage/committer.rs</b></font>
<font color="#EF2929"><b>+++ b/core/src/banking_stage/committer.rs</b></font>
<font color="#EF2929"><b>diff --git a/core/src/banking_stage/consumer.rs b/core/src/banking_stage/consumer.rs</b></font>
<font color="#EF2929"><b>+++ b/core/src/banking_stage/consumer.rs</b></font>
<font color="#EF2929"><b>+    pub fn </b></font>consume_buffered_packets(
<font color="#EF2929"><b>+    fn </b></font>do_process_packets(
<font color="#EF2929"><b>+    fn </b></font>process_packets_transactions(
<font color="#EF2929"><b>+    fn </b></font>process_transactions(
<font color="#EF2929"><b>+    pub fn </b></font>process_and_record_transactions(
<font color="#EF2929"><b>+    fn </b></font>execute_and_commit_transactions_locked(
<font color="#EF2929"><b>+    fn </b></font>accumulate_execute_units_and_time(execute_timings: &amp;ExecuteTimings) -&gt; (u64, u64) {
<font color="#EF2929"><b>+    fn </b></font>filter_pending_packets_from_pending_txs(
<font color="#EF2929"><b>+    fn </b></font>prepare_filter_for_pending_transactions(
<font color="#EF2929"><b>+    fn </b></font>filter_valid_transaction_indexes(valid_txs: &amp;[TransactionCheckResult]) -&gt; Vec&lt;usize&gt; {
<font color="#EF2929"><b>+    fn </b></font>sanitize_transactions(txs: Vec&lt;Transaction&gt;) -&gt; Vec&lt;SanitizedTransaction&gt; {
<font color="#EF2929"><b>+    fn </b></font>execute_transactions_with_dummy_poh_service(
<font color="#EF2929"><b>+    fn </b></font>generate_new_address_lookup_table(
<font color="#EF2929"><b>+    fn </b></font>store_address_lookup_table(
<font color="#EF2929"><b>+    fn </b></font>setup_conflicting_transactions(
<font color="#EF2929"><b>+    fn </b></font>test_bank_process_and_record_transactions() {
<font color="#EF2929"><b>+    fn </b></font>test_bank_process_and_record_transactions_all_unexecuted() {
<font color="#EF2929"><b>+    fn </b></font>test_bank_process_and_record_transactions_cost_tracker() {
<font color="#EF2929"><b>+    fn </b></font>test_bank_process_and_record_transactions_account_in_use() {
<font color="#EF2929"><b>+    fn </b></font>test_process_transactions_instruction_error() {
<font color="#EF2929"><b>+    fn </b></font>test_process_transactions_account_in_use() {
<font color="#EF2929"><b>+    fn </b></font>test_process_transactions_returns_unprocessed_txs() {
<font color="#EF2929"><b>+    fn </b></font>test_write_persist_transaction_status() {
<font color="#EF2929"><b>+    fn </b></font>test_write_persist_loaded_addresses() {
<font color="#EF2929"><b>+    fn </b></font>test_consume_buffered_packets() {
<font color="#EF2929"><b>+    fn </b></font>test_consume_buffered_packets_sanitization_error() {
<font color="#EF2929"><b>+    fn </b></font>test_consume_buffered_packets_interrupted() {
<font color="#EF2929"><b>+    fn </b></font>test_accumulate_execute_units_and_time() {
<font color="#EF2929"><b>+    fn </b></font>test_bank_prepare_filter_for_pending_transaction() {
<font color="#EF2929"><b>+    fn </b></font>test_bank_filter_valid_transaction_indexes() {
</pre>

**Review by ryoqun — APPROVED** [review]
lgtm; +/- 2k pr is nothing to fear about, if it's move-only and well-pr-commented. :)

File: core/src/banking_stage.rs [hunk 3]
```diff
@@ -1706,12 +982,6 @@ mod tests {
...
     }
 
-    fn sanitize_transactions(txs: Vec<Transaction>) -> Vec<SanitizedTransaction> {
-        txs.into_iter()
-            .map(SanitizedTransaction::from_transaction_for_tests)
-            .collect()
-    }
-
     #[test]
     fn test_bank_record_transactions() {
```
[the remark this answers is not in the document]
Comment (apfitzge) [reply core/src/banking_stage.rs:986]: You caught me, this does belong in `PohRecorder`, but there are some helper functions it uses and other *actual* BankingStage tests use. Put off moving this and figuring out where we want the helpers to live until after other clean up (this PR) where I already knew where I wanted things 😄

**Pull request merged**